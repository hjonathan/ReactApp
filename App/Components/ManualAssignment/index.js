import React, {Component} from "react";
import {View, Text, Picker} from "react-native";
import I18n from "../../I18n";
import Styles from "./styles";

/**
 * Class ItemRoute
 * Example:
 * import ManualAssignment from '../ManualAssignment';
 * render() {
 *      return(
 *          <ManualAssignment data={object}/>
 *      ),
 * }
 */
class ManualAssignment extends Component {
    /**
     * Constructor ItemList
     * @param props
     */
    constructor (props) {
        super(props);
        this.state = {
            data: (props && props.data) || [],
            items: [],
            itemValue: 0
        };
        this.onValueChange = this.onValueChange.bind(this);
    }

    /**
     * Component will be mounted.
     */
    componentWillMount () {
        const serviceItems = this.state.data.users.map((s, i) => {
            let {taskId} = this.state.data,
                itemValue = `${taskId}/${s.userId}`;
            if (i === 0) {
                this.props.addUserId({[taskId]: s.userId});
                this.setState({itemValue});
            }
            return (
                <Picker.Item key={s.userId} value={itemValue} label={s.userFullName} />
            );
        });
        this.setState({items: serviceItems});
    }

    /**
     * Change with select value in state
     * @param itemValue
     * @param itemIndex
     */
    onValueChange (itemValue, itemIndex) {
        let taskId,
            userId;
        [taskId, userId] = itemValue.split("/");
        this.setState({itemValue});
        this.props.addUserId({[taskId]: userId});
    }

    /**
     * Render
     * @returns {*}
     */
    render () {
        return (
            <View style={Styles.mainContainer} elevation={3}>
                <Text style={Styles.text}>
                    {I18n.t("task_name_label_user_route_screen")}{" "}
                    {this.state.data.taskName}
                </Text>
                <View style={Styles.firstSection}>
                    <View>
                        <Picker
                            selectedValue={this.state.itemValue}
                            onValueChange={this.onValueChange}
                        >
                            {this.state.items}
                        </Picker>
                    </View>
                </View>
                <View style={Styles.dividerLine} />
            </View>
        );
    }
}

export default ManualAssignment;
