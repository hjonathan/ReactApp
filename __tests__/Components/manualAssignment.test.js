import 'react-native';
import React from 'react';
import ManualAssignment from '../../App/Components/ManualAssignment';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Test of new item card for list', () => {
    let container;
    const data = {
                users: [
                    {
                        userId: "00000001",
                        userFullName: "Test 0001"
                    }
                ],
                taskId: "10000000",
            },
        addUserId = jest.fn();

    beforeEach(() => {
        container = shallow(
            <ManualAssignment
                data={data}
                addUserId={addUserId}
            />
        );
    });

    it('render Item correctly', () => {
        expect(toJson(container)).toMatchSnapshot();
    });

    it('Render the connected(SMART) component', () => {
        expect(container.length).toEqual(1)
    });

    it('Check Prop matches with data', () => {
        expect(container.instance().state.data).toEqual(data);
    });

    it('addUserId function', () => {
        expect(container.instance().props.addUserId()).toMatchSnapshot();
    });
    it('onValueChange function', () => {
        expect(container.instance().onValueChange("00001/10000")).toMatchSnapshot();
        expect(container.instance().state.itemValue).toEqual("00001/10000");
    });
});
