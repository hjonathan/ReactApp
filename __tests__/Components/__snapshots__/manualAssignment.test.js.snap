// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of new item card for list addUserId function 1`] = `undefined`;

exports[`Test of new item card for list onValueChange function 1`] = `undefined`;

exports[`Test of new item card for list render Item correctly 1`] = `
<Component
  elevation={3}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 14,
        "marginHorizontal": "10%",
        "textAlign": "center",
      }
    }
  >
    task_name_label_user_route_screen
     
  </Text>
  <Component>
    <Component>
      <Picker
        mode="dialog"
        onValueChange={[Function]}
        selectedValue="10000000/00000001"
      >
        <PickerItem
          key="00000001"
          label="Test 0001"
          value="10000000/00000001"
        />
      </Picker>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "#BABABA",
        "flex": 1,
        "height": 1,
        "marginHorizontal": 8,
      }
    }
  />
</Component>
`;
